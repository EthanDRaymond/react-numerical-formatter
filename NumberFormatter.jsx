import React from 'react';
import PropTypes from 'prop-types';

export default class NumberFormatter extends React.Component {
	format (str) {
		const parts = [];
		for (let i = str.length; i > 0; i -= 3) {
			parts.push(str.substring(i - 3, i));
		}

		parts.reverse();

		return parts.join(",");
	}

	render () {
		const string =  this.props.number.toString();

		if (string.includes(".")) {
			const parts = string.split(".");
			return `${this.format(parts[0])}.${parts[1]}`;
		} else {
			return this.format(string);
		}
	}
}

NumberFormatter.propTypes = {
	number: PropTypes.number.isRequired,
};
