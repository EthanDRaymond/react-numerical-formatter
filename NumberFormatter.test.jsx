import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import NumberFormatter from "./NumberFormatter.jsx";

configure({ adapter: new Adapter() });

describe(
	'Test Rendering.',
	() => {
		const NUMBERS = [];

		const MAX_DIGITS = 12;
		for (let i = 0; i <= MAX_DIGITS; i++) {
			const integer = Math.pow(10, i);
			const decimal = integer + 0.5;

			NUMBERS.push(integer);
			NUMBERS.push(decimal);
		}

		NUMBERS.push(123456789);

		NUMBERS.forEach(
			(number) => test(
				`Testing with ${number}.`,
				() => expect(shallow(<NumberFormatter number={number} />)).toMatchSnapshot(),
			),
		);
	},
);