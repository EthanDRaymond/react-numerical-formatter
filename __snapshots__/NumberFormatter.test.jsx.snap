// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rendering. Testing with 1. 1`] = `"1"`;

exports[`Test Rendering. Testing with 1.5. 1`] = `"1.5"`;

exports[`Test Rendering. Testing with 10. 1`] = `"10"`;

exports[`Test Rendering. Testing with 10.5. 1`] = `"10.5"`;

exports[`Test Rendering. Testing with 100. 1`] = `"100"`;

exports[`Test Rendering. Testing with 100.5. 1`] = `"100.5"`;

exports[`Test Rendering. Testing with 1000. 1`] = `"1,000"`;

exports[`Test Rendering. Testing with 1000.5. 1`] = `"1,000.5"`;

exports[`Test Rendering. Testing with 10000. 1`] = `"10,000"`;

exports[`Test Rendering. Testing with 10000.5. 1`] = `"10,000.5"`;

exports[`Test Rendering. Testing with 100000. 1`] = `"100,000"`;

exports[`Test Rendering. Testing with 100000.5. 1`] = `"100,000.5"`;

exports[`Test Rendering. Testing with 1000000. 1`] = `"1,000,000"`;

exports[`Test Rendering. Testing with 1000000.5. 1`] = `"1,000,000.5"`;

exports[`Test Rendering. Testing with 10000000. 1`] = `"10,000,000"`;

exports[`Test Rendering. Testing with 10000000.5. 1`] = `"10,000,000.5"`;

exports[`Test Rendering. Testing with 100000000. 1`] = `"100,000,000"`;

exports[`Test Rendering. Testing with 100000000.5. 1`] = `"100,000,000.5"`;

exports[`Test Rendering. Testing with 123456789. 1`] = `"123,456,789"`;

exports[`Test Rendering. Testing with 1000000000. 1`] = `"1,000,000,000"`;

exports[`Test Rendering. Testing with 1000000000.5. 1`] = `"1,000,000,000.5"`;

exports[`Test Rendering. Testing with 10000000000. 1`] = `"10,000,000,000"`;

exports[`Test Rendering. Testing with 10000000000.5. 1`] = `"10,000,000,000.5"`;

exports[`Test Rendering. Testing with 100000000000. 1`] = `"100,000,000,000"`;

exports[`Test Rendering. Testing with 100000000000.5. 1`] = `"100,000,000,000.5"`;

exports[`Test Rendering. Testing with 1000000000000. 1`] = `"1,000,000,000,000"`;

exports[`Test Rendering. Testing with 1000000000000.5. 1`] = `"1,000,000,000,000.5"`;
